### ISSUE TITLE
Error when install dotnet-sdk-2.1 on centos 7

### ISSUE THREAD

**Issue by nhuthan** [dotnet/core#1774]
I get this error:
```
Error: Package: dotnet-runtime-2.1-2.1.2-1.x86_64 (packages-microsoft-com-prod)
           Requires: dotnet-runtime-deps-2.1 >= 2.1.2
           Available: dotnet-runtime-deps-2.1-2.1.0-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-deps-2.1 = 2.1.0-1
           Available: dotnet-runtime-deps-2.1-2.1.1-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-deps-2.1 = 2.1.1-1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```
Please help

**Comment by nhuthan** [comment]
I tried today and everything installed successful on CentOS 7.4
I run `yum -y install deltarpm unzip zlib gettext libunwind libicu libcurl-devel openssl-devel libicu-devel` before install dotnet sdk.

**Comment by leecow** [comment]
@nhuthan did it install 2.1.2 or 2.1.1? `dotnet --info` will report everything that's installed.

**Comment by nhuthan** [comment]
Here is dotnet info:
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.302
 Commit:    9048955601

Runtime Environment:
 OS Name:     centos
 OS Version:  7
 OS Platform: Linux
 RID:         centos.7-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.302/

Host (useful for support):
  Version: 2.1.2
  Commit:  811c3ce6c0

.NET Core SDKs installed:
  2.1.302 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```
I run `yum install dotnet-sdk-2.1`

**Comment by leecow** [comment]
Looks like 2.1.2 is installing OK. Closing the issue.

**Comment by seamusic** [comment]
[root@Lennon-CentOS ~]# yum install dotnet-sdk-2.1
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Resolving Dependencies
--> Running transaction check
---> Package dotnet-sdk-2.1.x86_64 0:2.1.403-1 will be installed
--> Processing Dependency: dotnet-runtime-2.1 >= 2.1.5 for package: dotnet-sdk-2.1-2.1.403-1.x86_64
--> Processing Dependency: aspnetcore-runtime-2.1 >= 2.1.5 for package: dotnet-sdk-2.1-2.1.403-1.x86_64
--> Running transaction check
---> Package aspnetcore-runtime-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: dotnet-runtime-2.1 >= 2.1.6 for package: aspnetcore-runtime-2.1-2.1.6-1.x86_64
---> Package dotnet-runtime-2.1.x86_64 0:2.1.5-1 will be installed
--> Processing Dependency: dotnet-hostfxr-2.1 >= 2.1.5 for package: dotnet-runtime-2.1-2.1.5-1.x86_64
--> Processing Dependency: dotnet-runtime-deps-2.1 >= 2.1.5 for package: dotnet-runtime-2.1-2.1.5-1.x86_64
--> Running transaction check
---> Package aspnetcore-runtime-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: dotnet-runtime-2.1 >= 2.1.6 for package: aspnetcore-runtime-2.1-2.1.6-1.x86_64
---> Package dotnet-hostfxr-2.1.x86_64 0:2.1.5-1 will be installed
--> Processing Dependency: dotnet-host >= 2.1.5 for package: dotnet-hostfxr-2.1-2.1.5-1.x86_64
---> Package dotnet-runtime-deps-2.1.x86_64 0:2.1.5-1 will be installed
--> Running transaction check
---> Package aspnetcore-runtime-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: dotnet-runtime-2.1 >= 2.1.6 for package: aspnetcore-runtime-2.1-2.1.6-1.x86_64
---> Package dotnet-host.x86_64 0:2.1.6-1 will be installed
--> Finished Dependency Resolution
Error: Package: aspnetcore-runtime-2.1-2.1.6-1.x86_64 (packages-microsoft-com-prod)
           Requires: dotnet-runtime-2.1 >= 2.1.6
           Available: dotnet-runtime-2.1-2.1.0-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.0-1
           Available: dotnet-runtime-2.1-2.1.1-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.1-1
           Available: dotnet-runtime-2.1-2.1.2-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.2-1
           Available: dotnet-runtime-2.1-2.1.3-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.3-1
           Available: dotnet-runtime-2.1-2.1.4-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.4-1
           Installing: dotnet-runtime-2.1-2.1.5-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.5-1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest


I get the same error. why?

**Comment by mantzouranisc** [comment]
Following the instructions here:
https://www.microsoft.com/net/download/linux-package-manager/centos/runtime-current
I ran:
sudo rpm -Uvh https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm

And then:
sudo yum install aspnetcore-runtime-2.1.x86_64

I get:
```
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.wiuwiu.de
 * epel: mirror.wiuwiu.de
 * extras: mirror.ratiokontakt.de
 * updates: mirror.checkdomain.de
Resolving Dependencies
--> Running transaction check
---> Package aspnetcore-runtime-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: dotnet-runtime-2.1 >= 2.1.6 for package: aspnetcore-runtime-2.1-2.1.6-1.x86_64
--> Finished Dependency Resolution
Error: Package: aspnetcore-runtime-2.1-2.1.6-1.x86_64 (packages-microsoft-com-prod)
           Requires: dotnet-runtime-2.1 >= 2.1.6
           Available: dotnet-runtime-2.1-2.1.0-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.0-1
           Available: dotnet-runtime-2.1-2.1.1-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.1-1
           Available: dotnet-runtime-2.1-2.1.2-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.2-1
           Available: dotnet-runtime-2.1-2.1.3-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.3-1
           Available: dotnet-runtime-2.1-2.1.4-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.4-1
           Available: dotnet-runtime-2.1-2.1.5-1.x86_64 (packages-microsoft-com-prod)
               dotnet-runtime-2.1 = 2.1.5-1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

**Comment by karelz** [comment]
@leecow looks like 3 people hit the same issue ... any idea?

**Comment by karelz** [comment]
BTW: It would be best to file a separate issue as the latest error seems to be 2.1.6 specific. It would be great to confirm distro as well.
If anyone creates it, please link it here. Thanks!

**Comment by omajid** [comment]
@karelz It looks like aspnetcore-runtime 2.1.6 has landed in yum repos but the dependency dotnet-runtime 2.1.6 hasn't. dotnet-runtime 2.1.5 is found, but that's not sufficient for aspnetcore-runime 2.1.6. That's causing the error in both the logs above.

**Comment by karelz** [comment]
@omajid thanks for expalnation (still Linux newb here) - @leecow who can help to fix it? Looks like we forgot to publish all the right packages ...

**Comment by leecow** [comment]
We're in the middle of release 2.1.6 and looks like part of the publish didn't happen. Looking into it.

**Comment by leecow** [comment]
Confirmed 2.1.6 / SDK 2.1.500 now working on CentOS 7.

**Comment by Bambofy** [comment]
Still getting this error on Centos 6.10 (Final)
```
`[root@vps x]# sudo yum install dotnet-sdk-2.1
Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirror.hosting90.cz
 * epel: mirror.imt-systems.com
 * extras: mirror.hosting90.cz
 * nux-dextop: mirror.li.nux.ro
 * remi-php70: mirror.23media.de
 * remi-php72: mirror.23media.de
 * remi-safe: mirror.23media.de
 * updates: mirror.hosting90.cz
Resolving Dependencies
--> Running transaction check
---> Package dotnet-sdk-2.1.x86_64 0:2.1.500-1 will be installed
--> Processing Dependency: dotnet-runtime-2.1 >= 2.1.6 for package: dotnet-sdk-2                        .1-2.1.500-1.x86_64
--> Processing Dependency: aspnetcore-runtime-2.1 >= 2.1.6 for package: dotnet-s                        dk-2.1-2.1.500-1.x86_64
--> Running transaction check
---> Package aspnetcore-runtime-2.1.x86_64 0:2.1.6-1 will be installed
---> Package dotnet-runtime-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: dotnet-runtime-deps-2.1 >= 2.1.6 for package: dotnet-                        runtime-2.1-2.1.6-1.x86_64
--> Processing Dependency: dotnet-hostfxr-2.1 >= 2.1.6 for package: dotnet-runti                        me-2.1-2.1.6-1.x86_64
--> Running transaction check
---> Package dotnet-hostfxr-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: dotnet-host >= 2.1.6 for package: dotnet-hostfxr-2.1-                        2.1.6-1.x86_64
---> Package dotnet-runtime-deps-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: openssl-libs for package: dotnet-runtime-deps-2.1-2.1                        .6-1.x86_64
--> Running transaction check
---> Package dotnet-host.x86_64 0:2.1.6-1 will be installed
---> Package dotnet-runtime-deps-2.1.x86_64 0:2.1.6-1 will be installed
--> Processing Dependency: openssl-libs for package: dotnet-runtime-deps-2.1-2.1                        .6-1.x86_64
--> Finished Dependency Resolution
Error: Package: dotnet-runtime-deps-2.1-2.1.6-1.x86_64 (packages-microsoft-com-p                        rod)
           Requires: openssl-libs
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[root@vps x]#
```

**Comment by karelz** [comment]
2.1 supports only CentOS 7+: https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1-supported-os.md#linux
You versions (CentOS 6.10) is not supported.
